jerry-debugger/src/lib/protocol-handler.ts

```diff
@@ -466,6 +542,30 @@ export class JerryDebugProtocolHandler {
     return this.activeBreakpoints[breakpointId];
   }
 
+  evaluate(expression: string) {
+    if (!this.lastBreakpointHit) {
+      this.evalQueue.push(expression);
```

Comment:
If you set up watch expressions, it tries to evaluate them after each step command. However, it doesn't wait to receive a 'paused' event before doing so - I'm not quite sure why. But JrS doesn't allow that.

Hmm, looking more closely at the trace I guess it calls Debugger.evaluate() on the watch expressions right after the 'resume' event that comes in response to a step command. 

Then, after the next breakpoint is hit (i.e. step is finished), it calls Runtime.evaluateOnCallFrame with call frame of 0. So perhaps I could ignore Debugger.evaluate() for now and only respond to the call frame one. Also, I realize it would probably be better to report failure if they try to evaluate on a call frame other than 0, rather than just evaluate on frame 0 like now.

On relocating to cdt-controller.ts, I remember considering it but not giving it much thought, so you're probably right. I'll try to think about it a bit more. Or maybe given the above I can skip the queuing for now.

Reply:
After trying to ignore call frames other than 0, this is not actually great because it makes all expressions show up as <not available> even if they were defined at or above the requested call frame. So dropping that thought.